kotlinx-coroutines-core/jvm/test/ExecutorsTest.kt

```diff
@@ -32,14 +34,49 @@ class ExecutorsTest : TestBase() {
     }
 
     @Test
-    fun testToExecutor() {
+    fun testExecutorToDispatcher() {
         val executor = Executors.newSingleThreadExecutor { r -> Thread(r, "TestExecutor") }
         runBlocking(executor.asCoroutineDispatcher()) {
             checkThreadName("TestExecutor")
         }
         executor.shutdown()
     }
 
+    @Test
+    fun testConvertedDispatcherToExecutor() {
+        val executor: ExecutorService = Executors.newSingleThreadExecutor { r -> Thread(r, "TestExecutor") }
+        val dispatcher: CoroutineDispatcher = executor.asCoroutineDispatcher()
+        assertSame(executor, dispatcher.asExecutor())
+        executor.shutdown()
```

Comment:
Note: `shutdown` doesn't await actual shutdown

Reply:
That's Ok. Infrastrusture in `TestBase` awaits.

Reply:
It looks like it only awaits DefaultExecutor (and other dispatchers). You would need to register this out-of-band `Executor` with it somehow or put a factory method in `TestBase`.

Reply:
https://github.com/Kotlin/kotlinx.coroutines/blob/cd5e60e9f2470dced9687cd908b13c9c9aef887d/kotlinx-coroutines-core/jvm/test/Threads.kt#L47-L53

Reply:
Clever!